smart-scan/src/main/java/PreviousSeasons.java

```diff
@@ -130,170 +60,57 @@ public static int setPWP(String team, String sku) throws IOException {
 
         }
         catch (Exception e) {
-            return(0);
+            return(0.0);
         }
     }
-    public static int setPSP(String team, String sku) throws IOException {
-        ArrayList<String> previousdates = new ArrayList<>();
-        ArrayList<Date> previousDateList = new ArrayList<>();
-        Event eventThing = new Event(sku);
-        eventThing.setSeason();
-        Date currentDate = null;
-        String currentDateString = eventThing.getDatetime();
-        currentDateString = currentDateString.replace("%3A", "");
-        currentDateString = currentDateString.substring(0, currentDateString.indexOf("T"));
-
-        try
-        {
-            DateFormat formatter;
-            formatter = new SimpleDateFormat("yy-MM-dd");
-            currentDate = ((Date)formatter.parse(currentDateString));
-
-        }
-        catch (Exception e)
-        {}
-        URL url = new URL("http://api.vexdb.io/v1/get_events?season=" + eventThing.getPreviousSeason(false) + "&team=" + team);
-        //System.out.println(url);
-        InputStreamReader reader = new InputStreamReader(url.openStream());
+    public static int setPWASP(String team, String sku, String option) throws IOException {
+        String oldSKU = getOldSKU(team, sku, false);
         JsonParser jsonParser = new JsonParser();
-        JsonArray results = (JsonArray) jsonParser.parse(reader).getAsJsonObject().get("result"); //idk what the hell this is
-
-        for(JsonElement result: results) {
-            JsonObject temp = result.getAsJsonObject();
-            String stuff = String.valueOf(temp.get("sku"));
-            stuff = stuff.replace("\"", "");
-            previousdates.add(stuff);
-
-        }
-        //System.out.println(previousdates);
-        for(int i = 0; i<previousdates.size(); i++) {
-            Event newEvent = new Event(previousdates.get(i));
-            newEvent.setSeason();
-            String time = newEvent.getDatetime();
-            time = time.replace("%3A", "");
-            time = time.substring(0, time.indexOf("T"));
-            //System.out.println(time);
-            try
-            {
-                DateFormat formatter;
-                formatter = new SimpleDateFormat("yy-MM-dd");
-                previousDateList.add((Date)formatter.parse(time));
-
-            }
-            catch (Exception e)
-            {}
-
-        }
-
-        //System.out.println(previousDateList);
-        //System.out.println(currentDate);
-        Date nearest = (getDateNearest(previousDateList, currentDate));
-        //System.out.println(nearest);
-        int indexSKUlookup = previousDateList.indexOf(nearest);
-        String oldSKU = (previousdates.get(indexSKUlookup));
-        URL url2 = new URL("http://api.vexdb.io/v1/get_rankings?team="+team + "&sku="+oldSKU);
-        //System.out.println(url2);
+        URL url2 = new URL("https://api.vexdb.io/v1/get_rankings?team="+team + "&sku="+oldSKU);
```

Comment:
I see you're using this URL repeatedly, you should make it a static final variable somewhere so you can easily change it.